garden-backend-service/src/api/tasks/tasks.py

```diff
@@ -0,0 +1,247 @@
+    DELETE = "delete"
+
+
+class SearchIndexUpdateError(Exception):
+    """Custom exception for search index update failures."""
+
+    def __init__(self, message, doi):
+        super().__init__(message)
+        self.doi = doi
+
+
+async def schedule_search_index_update(
+    update_operation: SearchIndexOperation,
+    garden: Garden,
+    settings: Settings,
+    db: AsyncSession,
+    auth_client: ConfidentialAppAuthClient,
+) -> None:
+    """Schedule an update on the search index for the given garden."""
+    try:
+        match update_operation:
+            case SearchIndexOperation.CREATE_OR_UPDATE:
+                await _create_or_update_on_search_index(
+                    garden, settings, db, auth_client
+                )
+            case SearchIndexOperation.DELETE:
+                await _delete_from_search_index(garden.doi, settings, db, auth_client)
+            case _:
+                pass
+
+        # Look for stale FailedSearchIndexUpdate records with this doi
+        update: FailedSearchIndexUpdate | None = await FailedSearchIndexUpdate.get(
+            db, doi=garden.doi
+        )
+        if update is not None:
+            # delete the record from the failed updates table
+            await db.delete(update)
+        await db.commit()
+    except SearchIndexUpdateError as e:
+        # The update failed, log the failure
+        failed_update = FailedSearchIndexUpdate(
+            doi=garden.doi,
+            operation_type=update_operation.value,
+            error_message=str(e),
+            retry_count=0,
+        )
+        await _log_failed_update(failed_update, db)
+        await db.commit()
+
+
+async def retry_failed_updates(
+    settings: Settings,
+    db_session: async_sessionmaker,
+    auth_client: ConfidentialAppAuthClient,
+) -> None:
+    """Periodically retry failed updates to the search index."""
+    while settings.SYNC_SEARCH_INDEX:
+        try:
+            logger.info("Synchronizing failed updates with search index...")
+            async with db_session() as db:
+                failed_updates = await _get_failed_updates(db, settings)
+                logger.info(f"{len(failed_updates)} records to update.")
+                successful, failed = await _process_failed_updates(
+                    failed_updates, db, settings, auth_client
+                )
+            logger.info(f"Successfully updated {successful} records")
+            logger.info(f"Failed to update {failed} records")
+        except asyncio.CancelledError:
+            logger.info("Synchronization loop canceled.")
+            break
+        except Exception as e:
+            logger.error(f"Error in synchronization loop: {e}")
+
+        await asyncio.sleep(settings.RETRY_INTERVAL)
+
+
+async def _process_failed_updates(
+    failed_updates: list[FailedSearchIndexUpdate],
+    db: AsyncSession,
+    settings: Settings,
+    auth_client: ConfidentialAppAuthClient,
+) -> (int, int):
```

Comment:
```suggestion
) -> tuple[int, int]:
```
I think this is the typing syntax for tuples, not tuple literals